.github/workflows/ci.yml

```diff
@@ -9,7 +9,10 @@ on:
   # See: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#push.
   push:
     branches:
-      - '**'
+      # Do not run branch pushes on forks, because they are typically redundant with
+      # pull requests to the main repo.
```

Comment:
In commit "ci: skip Github CI on branch pushes for forks" (1a06d80e7bb113b9588efc113262bdd20b60efd4)

I didn't really understand this comment until I read the PR description. Might be clearer as "Disable CI on branch pushes to forks. It will still run for pull requests. This prevents CI from running twice for typical pull request workflows."

Also, I'm not really sure how this change relates to the rest of the PR. It seems like it could be unrelated?

Reply:
Updated.

It serves the same purpose as `NO_BRANCH` on Cirrus, expect that I couldn't find a way to make that behaviour opt-in on Github CI.

Reply:
I moved 063ea4e0da9e1c06cc0f4aec97a03abac66ae59f to be right after 8259a13954878fd6188bb218c3b7320385516596 and expanded the commit message to explain why they don't use the same mechanism.